src/functions-and-arrays.js

```diff
@@ -1,37 +1,151 @@
 // Iteration #1: Find the maximum
+function maxOfTwoNumbers (num1, num2) {
+  if (num1>num2){
+    return num1
+  } else if (num1<num2){
+    return num2
+  } return num1
+}
+maxOfTwoNumbers(2,4)
+
 
 
 // Iteration #2: Find longest word
 const words = ['mystery', 'brother', 'aviator', 'crocodile', 'pearl', 'orchard', 'crackpot'];
+function findLongestWord (array) {
+  let longestWord = ""
+  if (array.length === 0) {return null}
+  for (let i=0; i<array.length; i++) {
+    if (array[i].length > longestWord.length){
+      longestWord =array[i] 
+    } 
+    }return longestWord
+  }
+findLongestWord(words)
+
 
-// Iteration #3: Calculate the sum
 
+// Iteration #3: Calculate the sum
 const numbers = [6, 12, 1, 18, 13, 16, 2, 1, 8, 10];
+function sumNumbers (array){
+let sum= 0
+for (let i=0; i<array.length; i++){
+sum += array[i]
+  }
+return sum
+}
+sumNumbers(numbers)
+
+
+
+//Bonus: Calculate the sum mixed array
+const mixedArr = [6, 12, 'miami', 1, true, 'barca', '200', 'lisboa', 8, 10];
+function sum(array){
+  let result = 0;
+  for (let i=0; i<array.length; i++) {
+    if (typeof(array[i]) === 'string'){
+     result += array[i].length 
+    } else if (typeof(array[i]) === null || typeof(array[i]) === NaN || typeof(array[i]) === false || typeof(array[i]) === undefined) {
+     result += 0 
+    } else if (typeof(array[i]) === true){
+     result += 1 
+    }
+    else {result += array[i];}
+  }
+  return result;
+}
+sum(mixedArr)
+
+
 
 // Iteration #4: Calculate the average
 // Level 1: Array of numbers
 const numbersAvg = [2, 6, 9, 10, 7, 4, 1, 9];
+function averageNumbers(array){
+  if (array.length === 0) {return null}
+  let sum= 0
+  for (let i=0; i<array.length; i++){
+  sum += array[i]
+    }
+  return sum/array.length
+  }
+ averageNumbers(numbersAvg)
+
 
 // Level 2: Array of strings
 const wordsArr = ['seat', 'correspond', 'linen', 'motif', 'hole', 'smell', 'smart', 'chaos', 'fuel', 'palace'];
+function averageWordLength(array){
+  if (array.length === 0) {return null}
+  let sum= 0
+  for (let i=0; i<array.length; i++){
+  sum += array[i].length
+    }
+  return sum/array.length
+  }
+ averageNumbers(numbersAvg)
+
+
+
+//Bonus  average on mixed array 
+const mixedArr1 = [6, 12, 'miami', 1, true, 'barca', '200', 'lisboa', 8, 10];
+function avg(array){
+  let result = 0;
+  for (let i=0; i<array.length; i++) {
+    if (array.length === 0) {return null}
```

Comment:
Este` if` debe de ir fuera del `for`, por que es la función la que debe de devolver `null`, en este caso si pones un `return` dentro del `for`, salimos del bucle `for` pero la función sigue devolviendo algo:

```
const average= result/array.length
return average;
```

Que es `NaN`, no `null`